--- filesAdmin.py	2011-02-11 15:23:42.186821793 +0000
+++ filesAdmin_unattend.py	2011-02-11 15:23:42.186821793 +0000
@@ -11,6 +11,9 @@
 # IBM Corp.                                                         
 #                                                                   
 # ***************************************************************** 
+# This version hacked by Richard Clark <richard@fohnet.co.uk>
+# Added support for non-human interactions via serviceNum variable
+
 import lotusConnectionsCommonAdmin
 
 domainName = AdminControl.getDefaultDomain()
@@ -44,15 +47,19 @@
         i=i+1
         retry=1
 	while retry==1:
-	  print "Which service do you want to connect to?"
-	  response = sys.stdin.readline()
+      try:
+        serviceNum
+        response = serviceNum
+      except:
+	    print "Which service do you want to connect to?"
+        response = sys.stdin.readline()
 	  print ""
 	  try:
-	    j = int(response) - 1
-	    tt = svcs[j]
-	    retry=0
+	  j = int(response) - 1
+	  tt = svcs[j]
+	  retry=0
 	  except:
-	    print "Invalid selection, specify an index number between 1 and %i." % len(svcs)
+	  print "Invalid selection, specify an index number between 1 and %i." % len(svcs)
 	    
   nodeServer = svcs[j][svcs[j].find("node="):svcs[j].find(",", svcs[j].find("node="))] + ",*"
   FilesSchedulerServiceName = domainName+':name=FilesSchedulerMBean,type=LotusConnections,cell='+cellName+','+nodeServer
